run_common.go

```diff
@@ -1509,31 +1540,40 @@ func (b *Builder) runSetupRunMounts(mounts []string, sources runMountInfo, idMap
 				}
 			}
 		case "ssh":
-			mount, agent, err := b.getSSHMount(tokens, sshCount, sources.SSHSources, idMaps)
+			mount, agent, err := b.getSSHMount(tokens, len(agents), sources.SSHSources, idMaps)
 			if err != nil {
 				return nil, nil, err
 			}
 			if mount != nil {
 				finalMounts = append(finalMounts, *mount)
 				mountTargets = append(mountTargets, mount.Destination)
-				agents = append(agents, agent)
-				if sshCount == 0 {
+				if len(agents) == 0 {
 					defaultSSHSock = mount.Destination
 				}
-				// Count is needed as the default destination of the ssh sock inside the container is  /run/buildkit/ssh_agent.{i}
-				sshCount++
+				agents = append(agents, agent)
+			}
+		case "bind":
+			if bundleMountsDir == "" {
+				if bundleMountsDir, err = os.MkdirTemp(bundlePath, "mounts"); err != nil {
+					return nil, nil, err
+				}
 			}
-		case define.TypeBind:
-			mount, image, err := b.getBindMount(tokens, sources.SystemContext, sources.ContextDir, sources.StageMountPoints, idMaps, sources.WorkDir)
+			var mount *spec.Mount
+			mount, image, intermediateMount, overlayDir, err = b.getBindMount(tokens, sources.SystemContext, sources.ContextDir, sources.StageMountPoints, idMaps, bundleMountsDir)
```

Comment:
redeclaring `mount` isn't necessary on this branch; 1.27 has different types here is where that came from

Reply:
I just checked, vanilla 1.29 does have this structure so I'm thinking it should be preserved?

```
for _, mount := range mounts
  ...cut...
  switch mountType {
    ...cut...
    case define.TypeBind:
      mount, image, err := b.getBindMount(tokens, sources.SystemContext, sources.ContextDir, sources.StageMountPoints, idMaps, sources.WorkDir)
      ...cut...
```

Also, I don't know the golang scoping rules off-hand, I remember them being "weird". Regardless, let's see how this behaves in my next push.